### PR TITLE
feat: Pick out toolInstance from two listener

### PR DESCRIPTION
**Pull request by MMMMoriaty**

### CODE REVIEW

File: packages/lb-components/src/hooks/annotation.ts [hunk 1]
```diff
@@ -99,6 +102,10 @@ const useCustomToolInstance = ({ basicInfo }: ICustomToolInstanceProps = {}) =>
     });
   };
 
+  const clearToolInstance = () => {
+    Object.assign(toolInstanceRef.current, initialCustomToolInstance)
```
Comment (Kerwin-L) [inline packages/lb-components/src/hooks/annotation.ts:106]: You can clear more thoroughly like `toolInstanceRef.current = initialValue`

File: packages/lb-components/src/components/pointCloudView/index.tsx [hunk 2]
```diff
@@ -49,10 +49,21 @@ const PointCloudView: React.FC<IProps> = ({
...
 }) => {
-  const { globalPattern } = useContext(PointCloudContext);
+  const { globalPattern, setGlobalPattern } = useContext(PointCloudContext);
 
   const basicInfo = jsonParser(currentData.result);
-  const { toolInstanceRef } = useCustomToolInstance({ basicInfo });
+  const { toolInstanceRef, clearToolInstance } = useCustomToolInstance({ basicInfo });
+
+  useEffect(() => {
+    clearToolInstance()
```
Comment (Kerwin-L) [inline packages/lb-components/src/components/pointCloudView/index.tsx:58]: It's not a good place to clear data which depend on useEffect. You can clear data after patternChange

Comment (MMMMoriaty) [reply packages/lb-components/src/components/pointCloudView/index.tsx]: done

**Pull request merged**